core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGenePanelProfileMap.java

```diff
@@ -125,27 +126,31 @@ public void importData() throws Exception {
         }
 
         String line;
+        CancerStudy cs = DaoCancerStudy.getCancerStudyByStableId(cancerStudyStableId);
         while((line = buff.readLine()) != null) {
             List<String> data  = new LinkedList<>(Arrays.asList(line.split("\t")));
             String sampleId = data.get(sampleIdIndex);
-            Sample sample = genePanelRepository.getSampleByStableIdAndStudyId(sampleId, cancerStudyStableId);
+            Sample sample = DaoSample.getSampleByCancerStudyAndSampleId(cs.getInternalId() ,sampleId);
 
             data.remove((int)sampleIdIndex);
             for (int i = 0; i < data.size(); i++) {
-                List<GenePanel> genePanelList = genePanelRepository.getGenePanelByStableId(data.get(i));
-                if (genePanelList != null && genePanelList.size() > 0) {
-                    GenePanel genePanel = genePanelList.get(0);
-                    Map<String, Object> map = new HashMap<>();
-                    map.put("sampleId", sample.getInternalId());
-                    map.put("profileId", profileIds.get(i));
-                    map.put("panelId", genePanel.getInternalId());
-                    genePanelRepository.insertGenePanelSampleProfileMap(map);
+                GenePanel genePanel = DaoGenePanel.getGenePanelByStableId(data.get(i));
+                if (genePanel != null) {
+                    DaoSampleProfile.addSampleProfile(sample.getInternalId(), profileIds.get(i), genePanel.getInternalId());
                 }
                 else {
                     ProgressMonitor.logWarning("No gene panel exists: " + data.get(i));
                 }
+                if (DaoSampleProfile.sampleExistsInGeneticProfile(sample.getInternalId(), profileIds.get(i))) {
+                    samplesToDelete.add(sample.getInternalId());
+                    profilesToDelete.add(profileIds.get(i));                           
+                }
             }
         }
+        ProgressMonitor.setCurrentMessage("Deleting necessary records from sample_profile.");
+        DaoSampleProfile.deleteRecords(samplesToDelete, profilesToDelete);
+        ProgressMonitor.setCurrentMessage("Loading gene panel profile matrix data to database..");
+        MySQLbulkLoader.flushAll();
```

Comment:
Woops, I lost the code that did this when I realized there was an existing Dao class for SampleProfile.